app/src/js/App.js

```diff
@@ -42,29 +42,60 @@ let check = false;
 localStorage.setItem('mobile', check);
 
-// Wrapper for Main component to include routing
-const MainRoutes = () => {
+import { useState, useEffect } from 'react';
+
+const MainRoutes = ({ activeRoute }) => {
+  const history = useHistory();  // Use history for programmatic navigation
+  const [redirected, setRedirected] = useState(false); // State to track if redirection has occurred
+
+  // Define your routes in an array
+  const routes = [
+    { path: '/', component: Home, exact: true },
+    { path: '/error', component: Error },
+    { path: '/404', component: NotFound },
+    { path: '/requests', component: Requests },
+    { path: '/forms', component: Forms },
+    { path: '/daac/selection', component: FormRequest },
+    { path: '/form/questions/:id', component: FormQuestions },
+    { path: '/questions', component: Questions },
+    { path: '/users', component: Users },
+    { path: '/groups', component: Groups },
+    { path: '/workflows', component: Workflows },
+    { path: '/metrics', component: Metrics },
+    { path: '/roles', component: Roles },
+    { path: '/conversations/', component: Conversations },
+    { path: '/rules', component: Rules },
+    { path: '/modules', component: Modules },
+    { path: '/upload', component: Upload },
+    { path: '/test-api', component: TestApi },
+  ];
+
+  // Find the route that matches the activeRoute
+  const matchedRoute = routes.find(route => activeRoute.startsWith(route.path));
+
+  // Handle redirection initially
+  useEffect(() => {
+    if (matchedRoute && !redirected) {
+      // Redirect only if necessary, and activeRoute isn't already the current location
+      if (history.location.pathname !== activeRoute) {
+        history.push(activeRoute);
+      }
+      setRedirected(true);
+      localStorage.removeItem('redirectAfterLogin');
+    }
+  }, [activeRoute, matchedRoute, history, redirected]);
+
   return (
     <Main path='/'>
       <Switch>
-        <Route exact path='/' component={Home} />
-        <Route path='/error' component={Error} />
-        <Route path='/404' component={NotFound} />
-        <Route path='/requests' component={Requests} />
-        <Route path='/forms' component={Forms} />
-        <Route path="/daac/selection" component={FormRequest} />
-        <Route path="/form/questions/:id" component={FormQuestions} />
-        <Route path='/questions' component={Questions} />
-        <Route path='/users' component={Users} />
-        <Route path='/groups' component={Groups} />
-        <Route path='/workflows' component={Workflows} />
-        <Route path='/metrics' component={Metrics} />
-        <Route path='/roles' component={Roles} />
-        <Route path='/conversations/' component={Conversations} />
-        <Route path='/rules' component={Rules} />
-        <Route path='/modules' component={Modules} />
-        <Route path='/upload' component={Upload} />
-        <Route path='/test-api' component={TestApi} />
+        {routes.map(route => (
```

Comment:
Thanks for templating this! I think that should help out a lot more with development focusing more on the config aspect of this code rather than having to deal with the React aspects.